tools/pinning/pyproject.toml

```diff
@@ -34,8 +34,7 @@ certbot = {path = "../../certbot", extras = ["dev", "docs"]}
 acme = {path = "../../acme", extras = ["dev", "docs"]}
 
 # Extra dependencies
-# See https://github.com/certbot/certbot/issues/8425.
-mypy = "0.710"
+mypy = "0.812"
```

Comment:
No in fact it is the whole point to come back to the edge here! I removed the explicit pinning.

However, but it is slightly off topic, I feel that the dev extras defined in `certbot` project are less and less relevant in their current location. Indeed pylint, mypy et al. are not related to the `certbot` project itself, but to the entire `certbot` _repository_, since there are essentially used for `tox` and the other global test logics we have for Certbot. In my opinion it would make more sense to define them somewhere else. It could be `pyproject.toml`, but we would need to provide some logic to ensure we install them when necessary (eg. `tools/venv,py`, various setup in Azure jobs). Or it could be a dedicated local Python project even if it seems overkill. Anyway I do not have a clear proposition here, but I would be intersted to have your thoughts about it.

Reply:
I've been thinking about this myself a bit lately. The more specific thing I wanted to do was break out the tools we want installed in `tox` and CI like `pylint`, `mypy`, `pytest`, etc. from those we just want in the developer environment like `ipython`, `poetry`, etc.

To your point though, I agree what we have now is awkward, but I personally do like the simplicity of just installing `certbot[dev]` when needed. I think my ideal would essentially be an unpinned requirements file we could install with our constraints file when needed, but I don't immediately see a good way to tie that into the pinning system I started. I'm open to other ideas, but I think what I personally like best for now is to essentially keep things as they are, split up dev extras for performance, and maybe do something like put it all in `acme[dev]` and have other projects like `certbot[dev]` depend on `acme` as we need or want to keep all dev tools like that in once place.